lib/semant.ml

```diff
@@ -212,6 +216,7 @@ let check (globals, functions) =
         else raise (
             Failure ("return gives " ^ string_of_typ t ^ " expected " ^
                      string_of_typ func.rtyp ^ " in " ^ string_of_expr e))
+      | No_op -> SNo_op (* for the case where we only want if (..) {...} with no else block *)
```

Comment:
we should just do it the way microc did it to handle the if with no else block.

Reply:
microC does not handle that case
compactiC does. I just dont want it to be like we "copied it"

we could just add a check in the SIf section, where if expr2 is No_op then we add the block there.

I just dont want it to be close to what compactiC does